src/data/actions/accessibility.js

```diff
@@ -0,0 +1,41 @@
+import * as clientApi from '../api/client';
+import { accessibilityActions } from '../constants/actionTypes';
+
+export const submitAccessibilityFormSuccess = response => ({
+  type: accessibilityActions.ACCESSIBILITY_FORM_SUBMIT_SUCCESS,
+  statusCode: response.status,
+});
+
+export const submitAccessibilityFormRateLimitFailure = response => ({
+  type: accessibilityActions.ACCESSIBILITY_FORM_SUBMIT_RATE_LIMIT_FAILURE,
+  statusCode: 429,
+  failureDetails: response.detail,
+});
+
+export const submitAccessibilityForm = (formEmail, formFullName, formMessage) =>
+  dispatch =>
+    clientApi.postAccessibilityForm(formEmail, formFullName, formMessage)
+      .then((response) => {
+        // if 429 - Rate limit hit
+        // fetch response json to get more details
+        // else show user success message as server handles all other requests properly
+        if (response.status === 429) {
+          return response.json();
+        }
+
+        dispatch(submitAccessibilityFormSuccess(response));
+        return undefined;
```

Comment:
is this explicit `return` necessary? Doesn't `.then()` return a `Promise` with an `undefined` value if you don't specify a `return` value?

Reply:
Ah thanks! I wanted to discuss with the team about the proper way to handle this situation. I will come back to this once some other clean up has been done.